Packages/com.unity.inputsystem/InputSystem/Plugins/OnScreen/OnScreenControl.cs

```diff
@@ -208,13 +217,46 @@ protected void SentDefaultValueToControl()
             InputSystem.QueueEvent(m_InputEventPtr);
         }
 
+
+        // Used by PlayerInput auto switch for scheme to prevent using Pointer device. 
+        internal static bool HasAnyActive => s_nbActiveInstances != 0;
+        private static int s_nbActiveInstances = 0;
+
         protected virtual void OnEnable()
         {
+            ++s_nbActiveInstances;
             SetupInputControl();
+            if (m_Control == null)
+                return;
+            // if we are in single player and if it the first active switch to the target device.
+            if (s_nbActiveInstances == 1 &&
+                PlayerInput.isSinglePlayer)
+            {
+                var firstPlayer = PlayerInput.GetPlayerByIndex(0);
+                if (firstPlayer?.neverAutoSwitchControlSchemes == false)
+                {
+                    var devices = firstPlayer.devices;
+                    bool deviceFound = false;
+                    // skip is the device is already part of the current scheme
+                    foreach (var device in devices)
```

Comment:
I'm less confident with that part I choose to switch to target device in all case but maybe I just only try to move away from a pointer device. 🤔

Reply:
Yeah I started to think the same, that maybe the design error here is, first of all auto-switch with on-screen controls, but also that on screen controls rely on pointer instead of touch. Of course pointer is convenient when working in the editor but generally it feels like touch should be the underlying input here... In editor its possible to simulate touch from mouse so problem can be solved anyway....

Reply:
without isolation it's hard to reliably filter touch event with ugui event.